backend/utils/orders.js

```diff
@@ -30,4 +35,63 @@ async function findOrder(req, res, next) {
   next()
 }
 
-module.exports = { findOrder }
+const validPaymentStateTransitions = {
```

Comment:
Let's put all this new business logic code under logic/order and keep util for basic utility methods.